tests/preflight/preflight_test.go

```diff
@@ -31,222 +28,258 @@ var _ = Describe("Preflight Tests", func() {
 			It("Should set default logging level as INFO if incorrect logging level is provided", func() {
-				var byteData []byte
 				inputFlags := make(map[string]string)
 				copyMap(flagsMap, inputFlags)
 				inputFlags[logLevelFlag] = invalidLogLevel
-				runPreflightChecks(inputFlags)
-				byteData, err = getLogFileData(preflightLogFilePrefix)
+				cmdOut, err = runPreflightChecks(inputFlags)
 				Expect(err).To(BeNil())
-				outputLogData := string(byteData)
 
-				Expect(outputLogData).To(
+				Expect(cmdOut.Out).To(
 					ContainSubstring("Failed to parse log-level flag. Setting log level as INFO"))
-				assertSuccessfulPreflightChecks(inputFlags, outputLogData)
+				assertSuccessfulPreflightChecks(inputFlags, cmdOut.Out)
+			})
+		})
+
+		Context("Preflight run command cleanup on failure flag test cases", func() {
+			It("Should not clean resources when preflight check fails and cleanup on failure flag is set to false", func() {
+				inputFlags := make(map[string]string)
+				copyMap(flagsMap, inputFlags)
+				delete(inputFlags, cleanupOnFailureFlag)
+				inputFlags[localRegistryFlag] = invalidLocalRegistryName
+				cmdOut, err = runPreflightChecks(inputFlags)
+				Expect(err).To(BeNil())
+
+				By("Preflight pods should not be removed from cluster")
+				Eventually(func() int {
+					podList := unstructured.UnstructuredList{}
+					podList.SetGroupVersionKind(podGVK)
+					err = runtimeClient.List(ctx, &podList,
+						client.MatchingLabels(getPreflightResourceLabels("")), client.InNamespace(defaultTestNs))
+					Expect(err).To(BeNil())
+					return len(podList.Items)
+				}, timeout, interval).ShouldNot(Equal(0))
+
+				By("Preflight PVCs should not be removed from cluster")
+				Eventually(func() int {
+					pvcList := unstructured.UnstructuredList{}
+					pvcList.SetGroupVersionKind(pvcGVK)
+					err = runtimeClient.List(ctx, &pvcList,
+						client.MatchingLabels(getPreflightResourceLabels("")), client.InNamespace(defaultTestNs))
+					Expect(err).To(BeNil())
+					return len(pvcList.Items)
+				}, timeout, interval).ShouldNot(Equal(0))
 			})
 		})
 
 		Context("preflight run command, namespace flag test cases", func() {
 			It("Should perform preflight checks in default namespace if namespace flag is not provided", func() {
-				var byteData []byte
 				inputFlags := make(map[string]string)
 				copyMap(flagsMap, inputFlags)
 				delete(inputFlags, namespaceFlag)
-				runPreflightChecks(inputFlags)
-				byteData, err = getLogFileData(preflightLogFilePrefix)
+				cmdOut, err = runPreflightChecks(inputFlags)
 				Expect(err).To(BeNil())
-				outputLogData := string(byteData)
 
-				Expect(outputLogData).To(ContainSubstring("Using 'default' namespace of the cluster"))
+				Expect(cmdOut.Out).To(ContainSubstring("Using 'default' namespace of the cluster"))
+				assertSuccessfulPreflightChecks(inputFlags, cmdOut.Out)
+			})
+
+			It("Should perform preflight checks in namespace provided present on cluster", func() {
```

Comment:
update this test-case to negative scenario - if given ns is not present in cluster, test should fail.

And, all the tests in this suite should be run in `INSTALL_NAMESPACE`. To achieve this, I think you'll need to pass `-n` flag in every test input.